### PR TITLE
Webhooks

### PR DESCRIPTION
**Pull request by OrNovo**
Adding webhook resource and data-source

### CODE REVIEW

File: coralogix/clientset/clientset.go [hunk 1]
```diff
@@ -1,11 +1,33 @@
...
+
+var (
+	EnvToGrpcUrl = map[string]string{
+		"APAC1":   "ng-api-grpc.app.coralogix.in:443",
+		"APAC2":   "ng-api-grpc.coralogixsg.com:443",
+		"EUROPE1": "ng-api-grpc.coralogix.com:443",
+		"EUROPE2": "ng-api-grpc.eu2.coralogix.com:443",
+		"USA1":    "ng-api-grpc.coralogix.us:443",
+	}
+	EnvToRESTUrl = map[string]string{
```
Comment (oded-dd) [inline coralogix/clientset/clientset.go:15]: This is a problem. Let's discuss it.

**Review by oded-dd — CHANGES_REQUESTED** [review]

File: docs/resources/webhook.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    url  = "https://join.slack.com/example"
+  }
+}
+```
+
+### Webhook Webhook
+
+```hcl
+resource "coralogix_webhook" "custom_webhook" {
+  custom {
```
Comment (OrNovo) [inline docs/resources/webhook.md:31]: I changed from _webhook_ to _custom_

File: examples/webhook/main.tf [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    name = "opsgenie-webhook"
+    url  = "https://example-url.com/"
+  }
+}
+
+resource "coralogix_webhook" "demisto_webhook" {
+  demisto {
+    name    = "demisto-webhook"
+    payload = jsonencode({
+      "privateKey" : "d3bd1e44-5603-09db-52b6-cce4c850dc9e",
```
Comment (oded-dd) [inline examples/webhook/main.tf:188]: should not have privateKey (security issue)

File: examples/webhook/main.tf [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+  pager_duty {
+    name        = "pagerduty-webhook"
+    service_key = "service-key"
+  }
+}
+
+resource "coralogix_webhook" "email_group_webhook" {
+  email_group {
+    name   = "email-group-webhook"
+    emails = ["or.novog@gmail.com", "or.novogroder@coralgix.com"]
```
Comment (oded-dd) [inline examples/webhook/main.tf:156]: Put email that is not yours

File: examples/webhook/main.tf [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+provider "coralogix" {
+  #api_key = "<add your api key here or add env variable CORALOGIX_API_KEY>"
+  #env = "<add the environment you want to work at or add env variable CORALOGIX_ENV>"
+}
+
+resource "coralogix_webhook" "slack_webhook" {
+  slack {
+    name = "slack-webhook"
+    url  = "https://coralogix-dev.slack.com/archives/C03NAS1N7FC"
```
Comment (oded-dd) [inline examples/webhook/main.tf:18]: Lets make sure it is not valid channel

**Review by oded-dd — CHANGES_REQUESTED** [review]

File: docs/resources/alert.md [hunk 6]
```diff
@@ -33,7 +33,7 @@ resource "coralogix_alert" "standard_alert" {
   notification {
     recipients {
       emails      = ["user@example.com"]
-      webhook_ids = ["WebhookAlerts"] //change here for existing webhook from your account
```
Comment (OrNovo) [inline docs/resources/alert.md:36]: This is a breaking change, but I think it required.

File: examples/webhook/main.tf [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+          "text" : {
+            "integration_text" : "Security Incident",
+            "alert_application" : "$APPLICATION_NAME",
+            "alert_subsystem" : "$SUBSYSTEM_NAME",
+            "alert_severity" : "$EVENT_SEVERITY",
+            "alert_id" : "$ALERT_ID",
+            "alert_name" : "$ALERT_NAME",
+            "alert_url" : "$ALERT_URL",
+            "hit_count" : "$HIT_COUNT",
+            "alert_type_id" : "53d222e2-e7b2-4fa6-80d4-9935425d47dd"
```
Comment (oded-dd) [inline examples/webhook/main.tf:205]: I would show an example with `data`

**Review by oded-dd — APPROVED** [review]
Added small comment

**Review by oded-dd — APPROVED** [review]

**Pull request merged**